presto-hive/src/test/java/com/facebook/presto/hive/TestHiveRecoverableExecution.java

```diff
@@ -371,24 +427,10 @@ private void testRecoverableGroupedExecution(
 
             // kill worker2 only after the task has been scheduled
             TestingPrestoServer worker2 = workers.get(1);
-            Stopwatch stopwatch = Stopwatch.createStarted();
-            while (true) {
-                // wait for a while
-                sleep(500);
-
-                // if the task is already running - move ahead
-                if (hasTaskRunning(worker2)) {
-                    break;
-                }
-
-                // don't fail the test if task execution already finished
-                if (stopwatch.elapsed(SECONDS) > 5) {
-                    break;
-                }
-            }
+            sleep(1000);
             worker2.stopResponding();
 
-            assertEquals(result.get(60, SECONDS).getUpdateCount(), OptionalLong.of(expectedUpdateCount));
+            assertEquals(result.get(1000, SECONDS).getUpdateCount(), OptionalLong.of(expectedUpdateCount));
```

Comment:
question: It's 15 minutes. Why do we need such a large timeouts? I remember running these tests, and they were finishing withing ~20 seconds. Has it changed?

Reply:
Hasn't changed.  15 minutes there is unintentional.  I can fix it in a follow up.